deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1304 @@
+        var scriptsArray = CallHostFunctionNoLog(chakraDebug.JsDiagGetScripts);
+        scriptsArray.forEach(function(element, index, array) {
+          if (!_scripts[element.scriptId]) {
+            var scriptHandle = GetNextInternalHandle();
+            _scriptHandles[scriptHandle] = scriptHandle;
+            _scripts[element.scriptId] = new V8Script(element, scriptHandle);
+          }
+        });
+        return _scripts;
+      },
+      GetScript: function(scriptId) {
+        if (!_scripts[scriptId]) {
+          this.GetScripts();
+        }
+        return _scripts[scriptId];
+      },
+      IsScriptHandle: function(handle) {
+        return (_scriptHandles[handle]) ? true : false;
+      },
+      GetScriptFromHandle: function(handle) {
+        for (var i = 0; i < _scripts.length; ++i) {
+          if (_scripts[i] && _scripts[i].GetHandle() == handle) {
+            return _scripts[i];
+          }
+        }
+      },
+      GetScriptFromFileName: function(fileName) {
+        var scripts = this.GetScripts();
+        for (var i = 0; i < scripts.length; ++i) {
+          if (scripts[i] && scripts[i].IsSameFileName(fileName)) {
+            return scripts[i];
+          }
+        }
+      },
+      GetScriptFromMatchingFileName: function(regexToMatch) {
+        var scripts = this.GetScripts();
+        for (var i = 0; i < scripts.length; ++i) {
+          if (scripts[i] && scripts[i].GetFileName() &&
+            scripts[i].GetFileName().match(regexToMatch)) {
+            return scripts[i];
+          }
+        }
+      }
+    };
+
+    return {
+      Utility: Utility, BreakpointManager: BreakpointManager,
+      ScriptsManager: ScriptsManager
+    };
+  })();
+
+
+  var ExecutionStateType = {
+    Break: 0,
+    Async: 1
+  };
+
+  function ExecutionState(stateType, scriptId) {
+    this.type = stateType;
+    this.breakScriptId = scriptId;
+    this.frames = undefined;
+    this.objectEval = undefined;
+    this.objectProtoEval = undefined;
+    this.objectPrototypeEval = undefined;
+  }
+
+  ExecutionState.prototype.ChangeToBreak = function() {
+    this.type = ExecutionStateType.Break;
+  };
+
+  ExecutionState.prototype.IsAsyncBreak = function() {
+    return this.type == ExecutionStateType.Async;
+  };
+
+  ExecutionState.prototype.GetBreakScriptId = function() {
+    return this.breakScriptId;
+  };
+
+  ExecutionState.prototype.GetFrames = function() {
+    if (!this.frames || this.frames.length == 0) {
+      var _frames = [];
+      var stackTrace = CallHostFunction(chakraDebug.JsDiagGetStackTrace);
+      stackTrace.forEach(function(element, index, array) {
+        _frames.push(new V8Frame(element));
+      });
+
+      this.frames = _frames;
+    }
+    return this.frames;
+  };
+
+  ExecutionState.prototype.GetObjectEvalHandle = function() {
+    if (!this.objectEval) {
+      var objEval = CallHostFunction(chakraDebug.JsDiagEvaluate, 'Object', 0);
+      this.objectEval = objEval[1];
```

Comment:
>bjEval[1] [](start = 25, length = 9)

If i understand correctly, objEval[0] holds if call succeed or not. You never check that?